sources/packages/backend/libs/integrations/src/institution-integration/ece-integration/ece-response.processing.service.ts

```diff
@@ -18,38 +45,285 @@ export class ECEResponseProcessingService {
+          [AWARD_DETAILS_KEY]: [
+            {
+              payToSchoolAmount: eceDetailRecord.payToSchoolAmount,
+              isEnrolmentConfirmed: eceDetailRecord.isEnrolmentConfirmed,
+              enrolmentConfirmationDate:
+                eceDetailRecord.enrolmentConfirmationDate,
+            },
+          ],
+        };
+      } else {
+        const awardDetails =
+          disbursements[eceDetailRecord.disbursementIdentifier][
+            AWARD_DETAILS_KEY
+          ];
+        awardDetails.push({
+          payToSchoolAmount: eceDetailRecord.payToSchoolAmount,
+          isEnrolmentConfirmed: eceDetailRecord.isEnrolmentConfirmed,
+          enrolmentConfirmationDate: eceDetailRecord.enrolmentConfirmationDate,
+        });
+      }
+    }
+    return disbursements;
+  }
+
+  /**
+   * Validate the disbursement and application and
+   * update the enrolment status.
+   * @param disbursements disbursements to be processed,
+   * @param auditUser user who confirm or decline the enrolment.
+   * @param processSummary process summary.
+   * @returns count of disbursements processed, successfully updated, skipped and failed.
+   */
+  private async validateAndUpdateEnrolmentStatus(
+    disbursements: ECEDisbursements,
+    auditUserId: number,
+    processSummary: ProcessSummaryResult,
+  ): Promise<{
+    totalDisbursements: number;
+    disbursementsSuccessfullyProcessed: number;
+    disbursementsSkipped: number;
+    duplicateDisbursements: number;
+    disbursementsFailedToProcess: number;
+  }> {
+    const disbursementSchedules = Object.entries(disbursements);
+    const totalDisbursements = disbursementSchedules.length;
+    let disbursementsSuccessfullyProcessed = 0;
+    let disbursementsSkipped = 0;
+    let duplicateDisbursements = 0;
+    let disbursementsFailedToProcess = 0;
+    for (const [
+      disbursementScheduleId,
+      disbursementDetails,
+    ] of disbursementSchedules) {
+      const confirmedEnrolmentDetails = disbursementDetails.awardDetails.find(
+        (awardDetail) => awardDetail.isEnrolmentConfirmed,
+      );
+      try {
+        if (confirmedEnrolmentDetails) {
+          // Confirm enrolment.
+          const tuitionRemittance = disbursementDetails.awardDetails
+            .map((award) => award.payToSchoolAmount)
+            .reduce((accumulator, currentValue) => accumulator + currentValue);
+          await this.confirmationOfEnrollmentService.confirmEnrollment(
+            +disbursementScheduleId,
+            auditUserId,
+            tuitionRemittance,
+            {
+              enrolmentConfirmationDate:
+                confirmedEnrolmentDetails.enrolmentConfirmationDate,
+              applicationNumber: disbursementDetails.applicationNumber,
+            },
+          );
+          ++disbursementsSuccessfullyProcessed;
+          processSummary.summary.push(
+            `For disbursement ${disbursementScheduleId}, enrolment has been confirmed.`,
```

Comment:
I would prefer to keep it simple as `Disbursement ${disbursementScheduleId}, enrolment confirmed.`. Same for the others.